os/arch/arm/src/amebasmart/amebasmart_mipi.c

```diff
@@ -527,6 +524,26 @@ static int amebasmart_mipi_transfer(FAR struct mipi_dsi_host *dsi_host, FAR cons
 		rx_data_len = 0;
 	}
 	return OK;
+Fail_case:
+#ifdef CONFIG_PM
+	bsp_pm_domain_control(BSP_MIPI_DRV, 0);
+#endif
+#ifdef CONFIG_SMP
+	spin_unlock(&g_rtl8730e_config_dev_s_underflow);
+#endif
+	if (send_cmd_done == 0) {
+		sem_init(&g_send_cmd_done, 0, 0);
+	}
+	if (MIPI_LPTX_IS_READ(msg->type)) {
+		sem_init(&g_read_cmd_done, 0, 0);
+	}
+	if (msg->rx_buf && msg->rx_len > 0) {
+		memset(msg->rx_buf, 0, msg->rx_len);
+		rx_data_ptr = NULL;
+		rx_data_len = 0;
+	}
+	leave_critical_section(flags);
```

Comment:
The sequence in success case,
1. pm pause
2. spin lock for underflow
3. enter critical section
4. leave critical section
5. spin unlock
6. pm resume

but in failure
1. pm pause
2. spin lock for underflow
3. enter critical section
4. pm resume
5. spin unlock
6. leave critical section

Is it ok?

Reply:
you are right, i will change to the same sequence to avoid timing issue.